src/core/libmaven/jsonReports.h

```diff
@@ -1,32 +1,65 @@
-#ifndef JSONREPORTS_H
-#define JSONREPORTS_H
-
-#include <boost/algorithm/string.hpp>
-#include "standardincludes.h"
-
-// #include "../mzroll/tabledockwidget.h"
-
-using namespace std;
-
-class mzSample;
-class EIC;
-class PeakGroup;
-class MavenParameters;
-
-class JSONReports{
-
-public:
-    MavenParameters* mavenParameters;
-    JSONReports();
-    JSONReports(MavenParameters* _mp, bool pollyUpload = false);
-    ~JSONReports();
-    void saveMzEICJson(string filename,vector<PeakGroup> allgroups,vector<mzSample*> vsampleNames);
-    void writeGroupMzEICJson(PeakGroup& grp,ofstream& myfile, vector<mzSample*> vsampleNames);
-    string sanitizeJSONstring(string s);
-    float outputRtWindow = 2.0;
-private:
-    bool _uploadToPolly;
-};
-
-
-#endif
+#ifndef JSONREPORTS_H
+#define JSONREPORTS_H
+
+#include <boost/algorithm/string.hpp>
+#include "standardincludes.h"
+
+using namespace std;
+
+class mzSample;
+class EIC;
+class PeakGroup;
+class MavenParameters;
+
+class JSONReports{
+
+public:
+    JSONReports();
+    JSONReports(MavenParameters* mp, bool pollyUpload = false);
+    ~JSONReports();
+
+    /**
+     * @brief save Stores the compounds information in a json file.
+     * @param filename Output filename.
+     * @param allgroups Formed after processing the samples.
+     * @param vsampleNames  vector of samples uploaded.
+     */
+    void save(string filename, vector<PeakGroup> allgroups, vector<mzSample*> vsampleNames);
+
+private:  
+    /**
+     * @brief _writeGroup write specific group information to the file.
+     * @param grp Group to be written.
+     * @param myfile filename.
+     */
+    void _writeGroup(PeakGroup& grp, ofstream& myfile);
+
+    /**
+     * @brief _writeGroup write peak information to the file.
+     * @param grp PeakGroup to be written.
+     * @param myfile filename.
+     * @param samples uploaded.
+     */
+    void _writePeak(PeakGroup& grp, ofstream& myfile, vector<mzSample*> vsampleNames);
+
+    /**
+     * @brief _writeEIC EIC infomation exported to csv file.
+     * @param grp
+     * @param myfile
+     */
+    void _writeEIC(PeakGroup& grp, ofstream& myfile, mzSample*);
+
+    /**
+     * @brief _writeCompoundLink writes Compound Link of group
+     * @param grp
+     * @param myfile
+     */
+    void _writeCompoundLink(PeakGroup& grp, ofstream& myfile);

```

Comment:
Please don't name variables as `myfile`. It seems too toy-ish for professional code.